src/utils/hooks/useFetch.ts

```diff
@@ -6,39 +6,42 @@ import { setNotification } from '../../redux/actions/notificationActions';
 import { NotificationVariant } from '../../types/notification.types';
 import { apiRequest, authenticatedApiRequest } from '../../api/apiRequest';
 
-export const useFetch = <T>(url: string, withAuthentication = false): [T | undefined, boolean] => {
+export const useFetch = <T>(requestUrl: string, withAuthentication = false): [T | undefined, boolean] => {
   const dispatch = useDispatch();
   const { t } = useTranslation('feedback');
   const [data, setData] = useState<T>();
   const [isLoading, setIsLoading] = useState(false);
   const cancelToken = useCancelToken();
 
-  const fetchData = useCallback(async () => {
-    setIsLoading(true);
-    const fetchedData = withAuthentication
-      ? await authenticatedApiRequest<T>({ url, cancelToken })
-      : await apiRequest<T>({ url, cancelToken });
+  const fetchData = useCallback(
+    async (url: string) => {
+      setIsLoading(true);
+      const fetchedData = withAuthentication
+        ? await authenticatedApiRequest<T>({ url, cancelToken })
+        : await apiRequest<T>({ url, cancelToken });
 
-    if (fetchedData) {
-      if (fetchedData.error) {
-        dispatch(
-          setNotification(
-            t('error.fetch', { resource: url, interpolation: { escapeValue: false } }),
-            NotificationVariant.Error
-          )
-        );
-      } else if (fetchedData.data) {
-        setData(fetchedData.data);
+      if (fetchedData) {
+        if (fetchedData.error) {
+          dispatch(
+            setNotification(
+              t('error.fetch', { resource: url, interpolation: { escapeValue: false } }),
+              NotificationVariant.Error
+            )
+          );
+        } else if (fetchedData.data) {
+          setData(fetchedData.data);
+        }
       }
-    }
-    setIsLoading(false);
-  }, [dispatch, t, cancelToken, url, withAuthentication]);
+      setIsLoading(false);
```

Comment:
Hadde det vært mulig å wrappe dette inn i try-catch og finally block?

Reply:
Godt poeng, men vi tok kanskje et litt uheldig valg når vi bestemte oss for å lage en egen respons handler i `apiRequest.ts` i stedet for å bruke det som er innebygd i Axios (`Axios.get<MyObject>)(xxx)`), slik at `error` vi får her er ikke en exception men bare en property.

Vi har nylig begynt å stort sett bruke `useFetch` i stedet for dedikerte `useFetchRegistrations`, `useFetchProjects`, etc, så det vil forhåpentligvis gjøre det enklere å skrive seg over til axios sin måte å gjøre det på sikt, siden det da bare vil påvirke `useFetch`.